File: DAILY_CHALLENGE/MARCH_2024/26_03_2024.py
class Solution:

    def firstMissingPositive(self, nums: List[int]) -> int:
        INT_MAX=1e9+7
        n=len(nums)
        for i in range(0,n):
            if(nums[i]<=0):
                nums[i]=INT_MAX-1

        for i in range(0,n):
            index=abs(nums[i])-1
            if(index<n and nums[index]>0):
                nums[index]*=-1

        for i in range(0,n):
            if(nums[i]>0):
                return i+1
        
        return n+1

File: DAILY_CHALLENGE/MARCH_2024/26_03_2024.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        int n=nums.size();
        for(int i=0;i<n;i++)
        {
            if(nums[i]<=0)
            {
                nums[i]=INT_MAX-1;
            }
        }
        for(int i=0;i<n;i++)
        {
            int index=abs(nums[i])-1;
            if(index<n && nums[index]>0) 
            {
                nums[index]*=-1;
            }
            
        }
        for(int i=0;i<n;i++)
        {
            if(nums[i]>0)
            {
                return i+1;
            }
        }
        return n+1;
      
        
    }
};
